### PR TITLE
Throttler

### PR DESCRIPTION
**Pull request by mattford63**

### CODE REVIEW

File: src/kixi/event_fixer.clj [hunk 1]
```diff
@@ -194,6 +226,31 @@
...
+  [xf]
+  (let [a (atom 0)]
+    (fn
+      ([] (xf))
+      ([acc]
+       (xf acc))
+      ([acc x]
+       (when (= 1 (mod @a 2))
+         (println "Having a little sleep")
+         (Thread/sleep 1000))
```
Comment (otfrom) [inline src/kixi/event_fixer.clj:243]: why do you need to sleep if the problem is `MAX_CONNECTION`s?

File: src/kixi/event_fixer.clj [hunk 2]
```diff
@@ -57,32 +91,30 @@
...
       (when-not (.exists local-file)
         (do (.createNewFile local-file)
-            (io/copy (:object-content s3-object)
-                     local-file)))
+            (let [s3-object (s3/get-object credentials :bucket-name s3-base-dir
+                                           :key (:key s3-object-summary))
+                  in (:input-stream s3-object)]
+              (io/copy in
+                       local-file)
+              (. in close))))
```
Comment (otfrom) [inline src/kixi/event_fixer.clj]: use `with-open` instead to make sure things get closed if there is an exception (or put this in a `finally`)

**Review by otfrom — CHANGES_REQUESTED** [review]

File: src/kixi/event_fixer.clj (hunk 1 above)
Comment (mattford63) [reply src/kixi/event_fixer.clj:243]: We don't - this is a throttler we don't actually use now.  But handy to have about to plug into the transducer if we ever do so I left it.

File: src/kixi/event_fixer.clj (hunk 2 above)
Comment (mattford63) [reply src/kixi/event_fixer.clj]: Done

File: src/kixi/event_fixer.clj (hunk 1 above)
Comment (otfrom) [reply src/kixi/event_fixer.clj:243]: Is it worth moving into a gist then rather than being a distraction here?

**Review by otfrom — APPROVED** [review]

**Pull request merged**